tools/init-gke.sh

```diff
@@ -0,0 +1,153 @@
+  echo "Options:"
+  echo "--no-cluster                   Do not create the cluster"
+  echo "--no-node-autoscale            Do not enable Node autoscaling"
+  echo "--no-nat                       Do not deploy the Cloud NAT (use only if you already have a Cloud NAT deployed as it's required by the private cluster to pull third party dependencies.)"
+  echo "--no-turbinia-sa               Do not create a Turbinia GCP service account (use only if disabling the Turbinia deployment in OSDFIR Infrastructure.)"
+  exit 1
+fi
+
+# Check if gcloud is installed
+if [[ -z "$( which gcloud )" ]] ; then
+  echo "gcloud CLI not found.  Please follow the instructions at "
+  echo "https://cloud.google.com/sdk/docs/install to install the gcloud "
+  echo "package first."
+  exit 1
+fi
+
+# Check configured gcloud project
+if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+  DEVSHELL_PROJECT_ID=$(gcloud config get-value project)
+  ERRMSG="ERROR: Could not get configured project. Please either restart "
+  ERRMSG+="Google Cloudshell, or set configured project with "
+  ERRMSG+="'gcloud config set project PROJECT' when running outside of Cloudshell."
+  if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+  echo "Environment variable \$DEVSHELL_PROJECT_ID was not set at start time "
+  echo "so attempting to get project config from gcloud config."
+  echo -n "Do you want to use $DEVSHELL_PROJECT_ID as the target project? (y / n) > "
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# TODO: Do real check to make sure credentials have adequate roles
+if [[ $( gcloud -q --project $DEVSHELL_PROJECT_ID auth list --filter="status:ACTIVE" --format="value(account)" | wc -l ) -eq 0 ]] ; then
+  echo "No gcloud credentials found.  Use 'gcloud auth login' and 'gcloud auth application-default login' to log in"
+  exit 1
+fi
+
+# Create the Turbinia service account
+if [[ "$*" != *--no-turbinia-sa* ]] ; then
+    # Enable IAM services
+    gcloud -q --project $DEVSHELL_PROJECT_ID services enable iam.googleapis.com
+    SA_MEMBER="serviceAccount:$SA_NAME@$DEVSHELL_PROJECT_ID.iam.gserviceaccount.com"
+    if [[ -z "$(gcloud -q --project $DEVSHELL_PROJECT_ID iam service-accounts list --format='value(name)' --filter=name:/$SA_NAME@)" ]] ; then
+      gcloud --project $DEVSHELL_PROJECT_ID iam service-accounts create "${SA_NAME}" --display-name "${SA_NAME}"
+      # Grant IAM roles to the service account
+      echo "Grant permissions on service account"
+      gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/compute.instanceAdmin'
+      gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/iam.serviceAccountUser'
+    else
+      echo "The GCP service account $SA_NAME already exists. Skipping creation..."
+    fi
+else
+  echo "--no-turbinia-sa specified. Skipping Turbinia GCP service account creation..."
```

Comment:
what is the use for this switch if we check on line 93 it is exists anyway?

Reply:
The second one is for if someone was completely trying to disable Turbinia from being deployed (and not create the GCP account) and the first one is the check for if it already exists to not trying to create it.